src/main/java/games/strategy/triplea/ui/menubar/LobbyMenu.java

```diff
@@ -391,8 +394,10 @@ private void updateAccountDetails() {
     if (returnValue == CreateUpdateAccountPanel.ReturnValue.CANCEL) {
       return;
     }
-    // TODO: Replace MD5Crypt.crypt with an encryption algorithm
-    final String error = manager.updateUser(panel.getUserName(), panel.getEmail(), MD5Crypt.crypt(panel.getPassword()));
+    final String error = Strings.emptyToNull(Strings
+        .nullToEmpty(manager.updateUser(panel.getUserName(), panel.getEmail(), MD5Crypt.crypt(panel.getPassword())))
+        + Strings.nullToEmpty(manager.updateUser(panel.getUserName(), panel.getEmail(),
+            RsaAuthenticator.hashPasswordWithSalt(panel.getPassword()))));
```

Comment:
Simple Answer: No.
Long Answer: This MD5Crypt hash essentially replaces the real password. If an attacker was to "listen" on network traffic, he could read the sent hash, something like `$1$sOmesALt$sOmeVerYOLDHAShinGfoRm` and save that in a table along with the username.
Of course this hash can't be entered as password, but the server (legacy behaviour we need to drop in the next breaking release) sends the exact same salt of the hash to the client, which the client uses to hash the entered password again, and sends the exact same hash to the server in order to verify it.
This hides the actual password securely, but creates a hashing which ist just a pseudo-random password, so the problem is just somewehere else.

Reply:
Actually lol, I mixed your original comment with the edit.
My answer refers to the question if we could just use MD5Crypt.hashpw

Reply:
Thanks for pointing out the defect in what I suggested.  What I should have written was

```java
BCrypt.hashpw(
    RsaAuthenticator.hashPasswordWithSalt(panel.getPassword()),
    BCrypt.gensalt())
```

Again, this is in reference to my original comment where I was just trying to find a way to avoid the hack in `UserManager#updateUser()`.

While sniffing the MD5Crypted password in this transaction opens the user up to an attack, I don't believe that's the case with the bcrypted password above.  Because of the way you designed the challenge-response system initiated by `LobbyLoginValidator`, what the attacker needs to know is `sha512("TripleA" + password)` in order to login as the user.  If instead we send `bcrypt(sha512("TripleA" + password))` in this transaction and the attacker sniffs it, the attacker wouldn't be able to back out the information required to login, right?

Reply:
@ssoloff That is correct. but inconsistent, and may introduce some other problems when refactoring the code anyways...
If we send the pre-hashed password, we'd need to check if it already has been bcrypted and do so if not...
Not sure if it would make sense to change it for now or to just refactor the whole code in a follow-up PR. The server side PR should be able to handle pre-hashed passwords in the updateUser section, so we can easily change that afterwards if refactoring does not work without breaking anything...